src/Microsoft.AspNetCore.HttpOverrides/IPNetwork.cs

```diff
@@ -1,12 +1,234 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Globalization;
 using System.Net;
+using System.Net.Sockets;
 
 namespace Microsoft.AspNetCore.HttpOverrides
 {
     public class IPNetwork
     {
+        static readonly char[] PrefixLengthSeparators = new char[] { '/' };
+
+        /// <summary>
+        /// All IPv4 addresses 0.0.0.0/0
+        /// </summary>
+        public static readonly IPNetwork AllIPv4 = new IPNetwork(IPAddress.Parse("0.0.0.0"), 0);
+
+        /// <summary>
+        /// IPv4 localhost range 127.0.0.0/8 (255.0.0.0)
+        /// </summary>
+        public static readonly IPNetwork IPv4Loopback = Parse("127.0.0.1/8");
+
+        /// <summary>
+        /// IPv4 link-local range 169.254.0.0/16 (255.255.0.0)
+        /// </summary>
+        public static readonly IPNetwork IPv4LinkLocal = Parse("169.254.0.0/16");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 range 10.0.0.0/8 (255.0.0.0)
+        /// </summary>
+        public static readonly IPNetwork IPv4Private10 = Parse("10.0.0.0/8");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 range 172.16.0.0/12 (255.240.0.0)
+        /// </summary>
+        public static readonly IPNetwork IPv4Private172 = Parse("172.16.0.0/12");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 range 192.168.0.0/16 (255.255.0.0)
+        /// </summary>
+        public static readonly IPNetwork IPv4Private192 = Parse("192.168.0.0/16");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 range 192.168.0.0/16 (255.255.0.0)
+        /// </summary>
+        public static readonly IPNetwork IPv4Multicast = Parse("224.0.0.0/4");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 Addresses 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16, link-local, and localhost
+        /// </summary>
+        public static readonly IPNetwork[] IPv4PrivateNetworks = new IPNetwork[]
```

Comment:
Exposing a `readonly` array allows any consumer to change/overwrite its elements. Is this acceptable here, in `All` and `PrivateNetworks` below?